website/docs/plugin/framework/functions/errors.mdx

```diff
@@ -0,0 +1,251 @@
+error. Only errors that pertain to a function argument will
+include this information.
+
+## How Errors Affect State
+
+**Returning function errors does not stop the state from being updated**.
+Terraform will still persist the returned state even when function errors
+are returned with it. This is to allow Terraform to persist the values that have
+already been modified when a resource modification requires multiple API
+requests or an API request fails after an earlier one succeeded.
+
+When returning function errors, we recommend resetting the state in the
+response to the prior state available in the configuration.
+
+## fwerror Package
+
+The framework provides the `fwerror` package for interacting with function errors.
+While the [Go documentation](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/fwerror)
+contains the complete functionality, this section will highlight the most
+common methods.
+
+### Working With Existing Function Errors
+
+#### Append
+
+When receiving `fwerror.FunctionErrors` from a function or method, such as
+`Run()`, these should typically be appended to the
+response function errors for the method. This can be accomplished with the
+[`Append(in ...FunctionError)` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/fwerror#FunctionErrors.Append).
+
+For example:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+    var stringArg string
+
+    resp.Errors.Append(req.Arguments.Get(ctx, &boolArg, &stringArg)...)
+
+    // ... other logic ...
+}
+```
+
+This method automatically ignores `nil` or empty slice function errors and
+deduplicates where possible.
+
+#### HasError
+
+The most typical form of function errors checking is ensuring that execution should
+not stop due to encountering an error, potentially causing further confusing
+errors or crashes. The [`HasError()` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/fwerror#FunctionErrors.HasError)
+will check each of the function errors for error severity and return true if found.
+
+For example:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+    var stringArg string
+
+	funcErrs := req.Arguments.Get(ctx, &boolArg, &stringArg)
+    resp.Errors.Append(funcErrs...)
+
+	if resp.Errors.HasError() {
+		return
+	}
+
+    // ... other logic ...
+}
+```
+
+In this example, you will note that we opted to check `resp.Errors`
+instead of `funcErrs`. Technically checking either is correct, however, checking
+the response function errors can help ensure that any response will include the
+expected function errors.
+
+### Creating Diagnostics
+
+When working with logic outside the framework, such as interacting with the
+vendor or `net/http` library to make the actual calls to manage infrastructure
```

Comment:
Let's avoid `net/http` here since we want to discourage those types of functions. 👍

Reply:
I've removed the reference to `net/http`.